src/System.Text.Json/tests/Serialization/TestClasses.cs

```diff
@@ -769,6 +769,48 @@ public void Verify()
         }
     }
 
+    public class TestClassWithStringToObjectDictionaries : ITestClass
+    {
+        public Dictionary<string, SimpleTestClass> MyDict { get; set; }
+
+        // TODO: Add support for classes as dictionary values in class-nested dicitonaries for the following types.
+        //public IDictionary<string, SimpleTestClass> MyIDict { get; set; }
+        //public IReadOnlyDictionary<string, SimpleTestClass> MyIReadOnlyDict { get; set; }
+        //public ImmutableDictionary<string, SimpleTestClass> MyImmutableDict { get; set; }
+        //public IImmutableDictionary<string, SimpleTestClass> MyIImmutableDict { get; set; }
+        //public ImmutableSortedDictionary<string, SimpleTestClass> MyImmutableSortedDict { get; set; }
+
+        public static readonly byte[] s_data = Encoding.UTF8.GetBytes(
+            @"{" +
+                @"""MyDict"":{" +
+                    @"""Key1"":" + SimpleTestClass.s_json + "," +
+                    @"""Key2"":" + SimpleTestClass.s_json +
+                @"}" +
+            @"}");
+
+        public void Initialize()
+        {
+            MyDict = new Dictionary<string, SimpleTestClass>();
+            {
+                SimpleTestClass obj = new SimpleTestClass();
+                obj.Initialize();
+                MyDict["Key1"] = obj;
+            }
+            {
+                SimpleTestClass obj = new SimpleTestClass();
+                obj.Initialize();
+                MyDict["Key2"] = obj;
+            }
+        }
+
+        public void Verify()
+        {
+            Assert.Equal(2, MyDict.Count);
+            MyDict["Key1"].Verify();
+            MyDict["Key2"].Verify();
+        }
+    }
+
```

Comment:
Some thoughts about additional tests that might be missing.
- Using a non-string key key - what's the failure mode.
- Using a user-defined implementation of `IImutableDictionary<>` as a property type - what's the behaviour?

Reply:
Added.